File: solutions/lua/raindrops/1/raindrops.lua
return function(n)
	local res = ""

	if n % 3 == 0 then
		res = res .. "Pling"
	end

	if n % 5 == 0 then
		res = res .. "Plang"
	end

	if n % 7 == 0 then
		res = res .. "Plong"
	end

	if not (n % 3 == 0 or n % 5 == 0 or n % 7 == 0) then
		res = res .. tostring(n)
	end

	return res
end

File: solutions/c/raindrops/1/raindrops.c
#include "raindrops.h"
#include <stdio.h>

#define BUFFER_LENGTH 16

static void aux(char **result, int n, int m, int *buffer_length, const char* txt) {
	if (n % m == 0)
		snprintf(*result, *buffer_length, "%s", txt),
		*result += 5, *buffer_length -=5;
}

void convert(char result[], int n) {
	int buffer_length = BUFFER_LENGTH;
	if (n % 3 && n % 5 && n % 7) {
		snprintf(result, buffer_length, "%d", n);
		return;
	}
	aux(&result, n, 3, &buffer_length, "Pling");
	aux(&result, n, 5, &buffer_length, "Plang");
	aux(&result, n, 7, &buffer_length, "Plong");
}
